merlin-sdk/src/main/java/gov/nasa/jpl/aerie/merlin/protocol/types/ActivityInstanceId.java

```diff
@@ -0,0 +1,3 @@
+package gov.nasa.jpl.aerie.merlin.protocol.types;
+
+public final record ActivityInstanceId(long id) {}
```

Comment:
Since this type is never passed from the driver to the model, it probably shouldn't be in `merlin.protocol`. Recommend moving it into `merlin.driver`.

Reply:
It's used in `merlin.server` and the adaptations as well. I suppose `merlin.server` _is_ dependent on `merlin.driver`, so it should be fine to move it there. I think this is a result of my weak understanding of the purpose of the protocol module. Your comment leads me to the conclusion that the protocol provides types for users of the Merlin system (like adaptations) (is that correct?). In that case, I agree with the idea of moving the type to merlin-driver.

Reply:
> Your comment leads me to the conclusion that the protocol provides types for users of the Merlin system (like adaptations) (is that correct?)

Kind of. The `merlin.protocol` package organizes the types and interfaces used to *communicate between* the driver and the model. If something is only used by the driver and clients of it, it belongs in `merlin.driver`. If something is only used by models, it belongs in either `merlin.framework` or the model's own package.

`merlin.protocol` is meant to be very small, as changes to it usually require models to be rebuilt and reuploaded. For example, Matt's work on activity output attributes requires that models be able to pass return values of an activty's effect model method back to the driver. There's no current capability for that transfer to happen, so the `merlin.protocol` has to be extended to support that exchange.

Reply:
In contrast, if we're just changing the types of some IDs used internally by the driver, but which the model never comes into contact with, then we shouldn't change the public `protocol`. It's an implementation detail of the driver.

Reply:
Alright, so I tried to make the change, but it doesn't work alone because the constraint module doesn't depend on merlin-driver. We'll need to ponder potential solutions and determine the best approach, as I'm out of time this week, and don't want to rush a decision just to get something in.

Reply:
This probably tells us that the constraints module is also slightly too coupled to other parts of the system; we definitely shouldn't put `ActivityInstanceId` in `protocol` just so that the constraints system can see it. 😁

Reply:
That's a good point. I'll just rollback the changes to the constraint module, leaving activity IDs as a primitive `long` there. We _could_ also provide a wrapper type in the constraints module, but I don't think that's worthwhile considering the much more confined scope of the constraint module's activity instance IDs.

Reply:
Nice, now things are pushed and the PR reflects these changes.

Reply:
:+1:

I can't mark this convo resolved, so you'll have to do that for me 😅